files/acl/restrict-unpublished-files.php

```diff
@@ -38,6 +38,30 @@ function check_file_visibility( $file_visibility, $file_path ) {
 		return FILE_IS_PUBLIC;
 	}
 
+	$attachment_restrict_file_meta = get_post_meta( $attachment_id, '_restrict_attachment', true );
+
+	// Updated conditional to use type specific empty check.
+	if ( strlen( $attachment_restrict_file_meta ) > 0 ) {
+		
+		// Configured in_array strict check.
+		// Reduced possible values to an official value to be checked.
+		if ( in_array( $attachment_restrict_file_meta, [ 1, '1' ], true ) ) {
```

Comment:
Isn't `$attachment_restrict_file_meta` truthy/falsy at this step?

Would this not be valid here?:

```php
if ( $attachment_restrict_file_meta ) {
```